File: views/src/components/user_auth.js
async function initAuth() {
  let signIn = document.getElementById("sign-in");
  let container = document.getElementById("clerk-container");
  await Clerk.load();
  signIn.addEventListener("click", () => {
    Clerk.openSignIn();
  });
  if (Clerk.session?.status === "active") {
    let user = document.createElement("button");
    container.appendChild(user);
    Clerk.mountUserButton(user, { afterSignOutUrl: window.location.origin });
  } else {
    signIn.style.display = null;
  }
}

document
  .getElementById("auth-script-tag")
  .addEventListener("load", initAuth, { once: true });

File: views/src/components/user_auth.rs
use maud::{html, Markup, PreEscaped};

const AUTH_SCRIPT: &str = include_str!("./user_auth.js");

pub fn user_auth() -> Markup {
    html! {
        // script
        //     #auth-script-tag
        //     defer
        //     src="/public/app_auth.js"
        //     type="text/javascript" {}
        button type="button" #sign-in .btn style="display: none" {
            "Sign in"
        }
        #clerk-container {}
        // script { (PreEscaped(AUTH_SCRIPT)) }
    }
}
